### ISSUE TITLE
URLs and course descriptions are downloaded every time

### ISSUE THREAD

**Issue by antonplagemann** [C0D3D3V/Moodle-Downloader-2#96]
## Avoid duplicates
* [x] Bug is not already reported in another issue (seems not related to #77)

## Describe the bug
In every run, the moodle downloader fetches some files again. These files are (in my case) either course descriptions (.md) or URLs (.URL). 
1. The .md files are downloaded again, because the content has changed
2. The .URL files are downloaded again because the file name has changed.

## Bug cause
This bug seems to be caused by moodle itself. It's very strange but this moodle instance does some random HTML-encoding of email addresses  in the mardown file, and some random URL-encoding of email addresses in the URLs.
For the .md files, you can see an example of this different encoding in the first picture.

## Possible fix
I implemented a quick fix for both cases (PR #97):

- For course descriptions (.md): unescape and unquote the `hashable_description` before hashing it.
- For URLs: unescape and unquote `fist_guess_filename` and `url` before creating a `File()` instance.

Theres is maybe a more beautiful way, for me this was the most simple fix :-)

## Steps to reproduce the issue
For me: just run moodle-dl again.
For others: maybe not reproducable because I dont know whats actually causing this specific behaviour on moodle.

### Screenshots
Different content on moodle website when reloading the page (bug is caused by strange moodle behaviour):
![grafik](https://user-images.githubusercontent.com/54081139/117570245-96cf1180-b0c9-11eb-9c55-62bd915dbaf3.png)

Terminal output:
![grafik](https://user-images.githubusercontent.com/54081139/117569416-db58ae00-b0c5-11eb-9425-5d4dcda00a82.png)

### Terminal protocol
```
[RUNNING IN DEBUG-MODE!]
Moodle Downloader starting...
Loading config...
Checking for changes for the configured Moodle-Account....
Downloading course information   7/  7 [WebEngineering ..|  5442]
Start downloading changed files...
All files stored in the Database!


4 changes found for the configured Moodle-Account.
eTeach_TINF19CS1
≠       \\?\C:\Users\xxx\Downloads\Moodle-Downloader-2\eTeach_TINF19CS1\Datenbanken II (Hr. Dr. xx)\Prüfungsleistung PräsentationGruppe von genau 2 Te.md


Social Engineering und Systemsicherheitsmanagement (T3INF9003)
≠       \\?\C:\Users\xxx\Downloads\Moodle-Downloader-2\Social Engineering und Systemsicherheitsmanagement (T3INF9003)\Organisatorisches\Kontaktdaten -Thomas x - d2xxx75@student.dhb.md
-       \\?\C:\Users\xxx\Downloads\Moodle-Downloader-2\Social Engineering und Systemsicherheitsmanagement (T3INF9003)\Organisatorisches\mailto -%64%32%308%3xxx@st%75%64en%74.%64%68bw-ma%6e%6eh%65%69%6d%2ede.URL
+       \\?\C:\Users\xxx\Downloads\Moodle-Downloader-2\Social Engineering und Systemsicherheitsmanagement (T3INF9003)\Organisatorisches\mailto -%64%32%30%38xxx@st%75%64en%74.d%68%62%77-%6d%61%6e%6e%68%65%69%6d%2ed%65.URL

All done. Exiting..
```
### MoodleDownloader.log
The logfile looks the same after each run.
```
2021-05-09 12:29:38  INFO  {main}  --- moodle-dl started ---------------------
2021-05-09 12:29:38  DEBUG  {main}  moodle-dl version: 2.1.2.6
2021-05-09 12:29:38  DEBUG  {main}  python version: 3.9.0
2021-05-09 12:29:38  DEBUG  {main}  Is ffmpeg available: False
2021-05-09 12:29:38  INFO  {main}  Debug-Mode detected. Errors will be re-risen.
2021-05-09 12:29:38  DEBUG  {main}  Loading config...
2021-05-09 12:29:38  DEBUG  {state_recorder}  Database Version: 4
2021-05-09 12:29:38  DEBUG  {main}  Checking for changes for the configured Moodle-Account....
2021-05-09 12:29:38  DEBUG  {moodle_service}  Fetching current Moodle State...
2021-05-09 12:29:38  DEBUG  {results_handler}  Detected moodle version: 2019052009
2021-05-09 12:29:38  DEBUG  {cookie_handler}  Testing cookies with this URL https://moodle.dhbw-mannheim.de/
2021-05-09 12:29:39  DEBUG  {cookie_handler}  Cookies are still valid
2021-05-09 12:29:45  DEBUG  {moodle_service}  Checking for changes...
2021-05-09 12:29:46  DEBUG  {cookie_handler}  Testing cookies with this URL https://moodle.dhbw-mannheim.de/
2021-05-09 12:29:46  DEBUG  {main}  Start downloading changed files...
2021-05-09 12:29:46  INFO  {console_service}  4 changes found for the configured Moodle-Account.
2021-05-09 12:29:46  DEBUG  {mail_service}  Mail-Notifications not configured, skipping.
2021-05-09 12:29:46  DEBUG  {telegram_service}  Telegram-Notifications not configured, skipping.
2021-05-09 12:29:46  DEBUG  {xmpp_service}  XMPP-Notifications not configured, skipping.
2021-05-09 12:29:46  DEBUG  {main}  All done. Exiting...
```

## Technical details
 - OS: Windows
 - Version: 1909

## Expected behavior
It should download these files only once.
